### PR TITLE
Release Candidate 1

### PR DESCRIPTION
**Pull request by johntwyman**
Minor cosmetic changes but predominantly adds in the code that sends a POST request to the Google Analytics REST API.

Reference is here: https://developers.google.com/analytics/devguides/collection/protocol/v1/reference

### CODE REVIEW

File: trackpaypal.php [hunk 1]
```diff
@@ -216,4 +216,60 @@ function trackpaypal_civicrm_postIPNProcess(&$IPNData) {
   // and transaction details to send to Google Analytics
   // via its REST interface
   watchdog('paypal', 'IPN payload: %payload', array('%payload' => print_r($IPNData, TRUE)), WATCHDOG_DEBUG);
+
+  // Retrieve extension settings
+  $result = civicrm_api3('setting', 'get', ['return' => ['trackpaypal_event_type'],]);
+  $event_type = $result['values']['1']['trackpaypal_event_type'];
```
Comment (seamuslee001) [inline trackpaypal.php]: @johntwyman what about trying `Civi::settings->get('trackpaypal_event_type')` and similar

Comment (johntwyman) [reply trackpaypal.php]: That looks more obvious. You happy to suggest the specific changes?

Comment (seamuslee001) [inline trackpaypal.php]:
```suggestion
  $event_type = Civi::settings->get('trackpaypal_event_type');
```
This is as per https://docs.civicrm.org/dev/en/latest/framework/setting/#setting-storage-and-retrieval

File: trackpaypal.php [hunk 2]
```diff
@@ -216,4 +216,58 @@ function trackpaypal_civicrm_postIPNProcess(&$IPNData) {
...
   watchdog('paypal', 'IPN payload: %payload', array('%payload' => print_r($IPNData, TRUE)), WATCHDOG_DEBUG);
+
+  // Retrieve extension settings
+    $event_type = Civi::settings->get('trackpaypal_event_type');
+    $tracking_code = Civi::settings->get('trackpaypal_tracking_code');
+
+  // Check the GA Code is of valid syntax
+  // If not we do nothing
+  if (!trackpaypal_isGACode($tracking_code)) {
+    watchdog('paypal', 'GA Code invalid: %code ', array('%code' => $tracking_code), WATCHDOG_DEBUG);
```
Comment (seamuslee001) [inline trackpaypal.php]: This is ok for Phase 1 but i think for Phase 2 looking at logging as per https://civicrm.org/blog/totten/the-static-is-dead-long-live-the-static might be better (more non CMS linked)

Comment (johntwyman) [reply trackpaypal.php]: Completely agree. `Civi::log()->error('GA Code invalid: {code}', array('code' => $tracking_code));` definitely looks smarter. Easy enough to update now I guess.

**Review by seamuslee001 — APPROVED** [review]

**Pull request merged**